lib/tasks/migrate_need_ids.rake

```diff
@@ -0,0 +1,51 @@
+desc 'One off task to migrate numeric need ids to links in the Publishing API'
+task migrate_need_ids: :environment do
+  results = ActiveRecord::Base.connection.execute("
+SELECT * FROM (
+  SELECT documents.content_id,
+         (SELECT need_ids
+          FROM editions
+          WHERE editions.document_id = documents.id
+          AND need_ids IS NOT NULL
+          ORDER BY editions.id DESC LIMIT 1) need_ids
+  FROM documents) results
+WHERE need_ids IS NOT NULL;")
+
+  not_found_need_ids = {}
+
+  results.each do |content_id, raw_need_ids|
+    need_ids = raw_need_ids.lines.map { |line|
+      line.gsub(/[^\d]/, '')
+    }.reject(&:empty?)
+
+    need_content_ids = need_ids.map do |need_id|
+      begin
+        Whitehall.need_api.content_id(need_id)
+      rescue GdsApi::HTTPNotFound
+        nil
+      end
+    end
+
+    mapping = need_ids.zip(need_content_ids).map { |n, c| "  #{n} -> #{c}" }.join("\n")
+    puts "#{content_id}:\n#{mapping}\n"
+
+    not_found_count = need_content_ids.count(nil)
+    not_found_need_ids[content_id] = not_found_count if not_found_count.nonzero?
```

Comment:
I'm finding this hard to read, partly because of the naming here but also I think it would be better to maybe split some of this rake task out into smaller methods, the naming of which would help make it easier to follow.